Assets/Scripts/Interactables/Earth/Resizable.cs

```diff
@@ -3,82 +3,58 @@
 
 namespace RogueApeStudios.SecretsOfIgnacios.Interactables.Earth
 {
-	internal class Resizable : EarthInteractable
-	{
-		[SerializeField] private Vector3 _initialSize;
-		[SerializeField] private Vector3 _shrinkSize;
-		[SerializeField] private Vector3 _growSize;
-
-		[SerializeField] private Transform _targetObject;
-
-		[SerializeField] private bool _shrunk;
-		[SerializeField] private bool _grown;
-
-		internal event Action onSizeChanged;
-		
-		public bool Shrunk => _shrunk;
-        public bool Grown => _grown;
-        
-		internal override void Awake()
-		{
-			//Sorry, but this was necessary for now, if you have something better, feel free to say so :)
-
-			base.Awake();
-			_initialSize = _targetObject.localScale;
-
-			if (_shrunk)
-				_targetObject.localScale = _shrinkSize;
-			else if (_grown)
-				_targetObject.localScale = _growSize;
-		}
-
-		internal override void Touched()
-		{
-			ResizeObject();
-		}
-
-		private void ResizeObject()
-		{
-			if (_isGrowSpellActive && !_grown)
-				GrowObject();
-
-			else if (_isShrinkSpellActive && !_shrunk)
-				ShrinkObject();
-		}
-
-		private void GrowObject()
-		{
-			if (_shrunk)
-			{
-				_targetObject.localScale = _initialSize;
-				_shrunk = false;
-			}
-			else
-			{
-				_targetObject.localScale = _growSize;
-				_grown = true;
-			}
-
-			_isGrowSpellActive = false;
-            onSizeChanged?.Invoke();
-		}
-
-		private void ShrinkObject()
-		{
-			if (_grown)
-			{
-				_targetObject.localScale = _initialSize;
-				_grown = false;
-			}
-
-			else
-			{
-				_targetObject.localScale = _shrinkSize;
-				_shrunk = true;
-			}
-
-			_isShrinkSpellActive = false;
+    internal class Resizable : EarthInteractable
+    {
+        [SerializeField] private Vector3 _shrinkSize;
+        [SerializeField] private Vector3 _defaultSize;
+        [SerializeField] private Vector3 _growSize;
+
+        [SerializeField] private Transform _targetObject;
+
+        [SerializeField] private ResizeState _currentState = ResizeState.Default;
+
+        internal event Action onSizeChanged;
+
+        public ResizeState CurrentState => _currentState;
+
+        internal override void Awake()
+        {
+            base.Awake();
+            SetObjectScale();
+        }
+
+        internal override void Touched()
+        {
+            ResizeObject();
+        }
+
+        private void ResizeObject()
+        {
+            if (_currentState == ResizeState.Shrunk)
```

Comment:
I recommend a switch case when working with enums, not a must. Also this could go directly in `Touched()` if it is the only thing there. Also Also can the resizable only grow at the moment?
```c#
switch (_currentState)
            {
                case ResizeState.Shrunk:
                    break;
                case ResizeState.Default:
                    break;
                case ResizeState.Grown:
                    break;
                default:
                    throw new NotImplementedException("State has not been implemented!");
            }
```